### PR TITLE
Update operator metadata bundling

### PR DESCRIPTION
**Pull request by lmm**
## Description

This PR updates the operator bundling used to generate metadata used to certify the operator. Previously we used the "packagemanifests" format but now we're using the "bundle" format.

Once this PR is merged, I'll need to pick these changes to 1.17, 1.15, and 1.13 and generate the bundles in the respective release branches.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -49,37 +49,33 @@ You should have no local changes and tests should be passing.
...
-4. Create the CSV bundle:
+   This step will create the bundle `bundle/1.13.1`.
 
-   ```
-   make gen-bundle
-   ```
+1. Login to our operator bundle project on connect.redhat.com
 
-   This step will add the CSV we generated earlier to `deploy/olm-catalog` and create a zip file in `build/_output/bundle/bundle.zip`.
+1. Tag and push the operator bundle image to connect.redhat.com
```
Comment (caseydavenport) [inline RELEASING.md:75]: Are there commands we can copy / paste to do this? Not obvious how this is done.

File: RELEASING.md [hunk 2]
```diff
@@ -49,37 +49,33 @@ You should have no local changes and tests should be passing.
...
 
-5. `git add deploy/olm-catalog`, commit the changes, and have the PR reviewed and merged into master.
+1. Create a PR, adding the new bundle and get it reviewed.
```
Comment (caseydavenport) [inline RELEASING.md:77]: Same here - is there an example PR we could link to? Where does the bundle get added?

File: hack/gen-bundle/get-manifests.sh [hunk 3]
```diff
@@ -27,17 +39,23 @@ fi
...
     # Download the installation CR so that the alm-examples annotation is generated.
-    curl ${CALICO_BASE_URL}/manifests/ocp/01-cr-installation.yaml --output ${DEPLOY_DIR}/01-cr-installation.yaml
+    curl ${CALICO_BASE_URL}/manifests/ocp/01-cr-installation.yaml --output ${BUNDLE_DEPLOY_DIR}/cr-installation.yaml
 }
 
 # Download the installation and tigerastatus CRDs.
 function downloadOperatorCRDs() {
-    curl ${CALICO_BASE_URL}/manifests/ocp/crds/01-crd-installation.yaml --output ${CSV_DIR}/operator.tigera.io_installations_crd.yaml
-    curl ${CALICO_BASE_URL}/manifests/ocp/crds/01-crd-tigerastatus.yaml --output ${CSV_DIR}/operator.tigera.io_tigerastatuses_crd.yaml
+    curl ${CALICO_BASE_URL}/manifests/ocp/crds/01-crd-installation.yaml --output ${BUNDLE_CRD_DIR}/operator.tigera.io_installations_crd.yaml
```
Comment (caseydavenport) [inline hack/gen-bundle/get-manifests.sh:56]: Don't we already have these locally in-repo?

File: RELEASING.md [hunk 4]
```diff
@@ -49,37 +49,33 @@ You should have no local changes and tests should be passing.
...
+If the release includes new Calico CRDs, add the new CRDs to `hack/gen-bundle/get-manifests.sh` and `config/manifests/bases/operator.clusterserviceversion.yaml`.
 
 ## Publishing a release on RH Catalog
 
 We currently only publish operator releases targeting Calico. If the release targets Calico, continue onto the following steps to generate the
-ClusterServiceVersion for it, and publish the release on the RH Catalog.
+operator bundle for it, and publish the release on the RH Catalog.
 
-1. After the semaphore job in the releasing steps is complete, and images have been tagged and pushed, checkout the tag you released.
+1. After the semaphore job in the releasing steps is complete, and images have been tagged and pushed, checkout the release branch containing the released tag.
```
Comment (caseydavenport) [inline RELEASING.md:59]: Why this change? Shouldn't we build the bundle from the precise tag?

File: RELEASING.md (hunk 1 above)
Comment (lmm) [reply RELEASING.md:75]: Edited it to add this

File: hack/gen-bundle/get-manifests.sh (hunk 3 above)
Comment (lmm) [reply hack/gen-bundle/get-manifests.sh:56]: I think at the time when I added the script the thinking was to get all of the crds from the same spot (our docs site).

File: RELEASING.md (hunk 4 above)
Comment (lmm) [reply RELEASING.md]: Hmm yeah this is odd, I reverted this phrasing. In practice the release branch head should be at that tag

File: hack/gen-bundle/get-manifests.sh (hunk 3 above)
Comment (caseydavenport) [reply hack/gen-bundle/get-manifests.sh:56]: I think that's fine if that's what we want to do - the RELEASING steps should be updated to make sure this only happens after the docs site has been properly updated (with time for docs.projectcalico.org to regenerate the site and push it to live)

**Review by caseydavenport — APPROVED** [review]
I won't pretend to full know how this all works but this seems sensible enough to me.
